ext/QuantumToolboxCairoMakieExt.jl

```diff
@@ -0,0 +1,215 @@
+- `projection::Union{Val,Symbol}`: Whether to plot the Wigner function in 2D or 3D. It can be either `Val(:two_dim)` or `Val(:three_dim)`, with `Val(:two_dim)` as default.
+- `location::Union{GridPosition,Nothing}`: The location of the plot in the layout. If `nothing`, the plot is created in a new figure. Default is `nothing`.
+- `colorbar::Bool`: Whether to include a colorbar in the plot. Default is `false`.
+- `kwargs...`: Additional keyword arguments to pass to the plotting function. 
+
+# Returns
+- `fig`: The figure object.
+- `ax`: The axis object.
+- `hm`: Either the heatmap or surface object, depending on the projection.
+
+!!! note "Import library first"
+    [`CairoMakie`](https://github.com/MakieOrg/Makie.jl/tree/master/CairoMakie) must first be imported before using this function.
+
+!!! warning "Beware of type-stability!"
+    If you want to keep type stability, it is recommended to use `Val(:two_dim)` and `Val(:three_dim)` instead of `:two_dim` and `:three_dim`, respectively. Also, specify the library as `Val(:CairoMakie)` See [this link](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-value-type) and the [related Section](@ref doc:Type-Stability) about type stability for more details.
+"""
+function QuantumToolbox.plot_wigner(
+    library::Val{:CairoMakie},
+    state::QuantumObject{DT,OpType};
+    xvec::Union{Nothing,AbstractVector} = LinRange(-7.5, 7.5, 200),
+    yvec::Union{Nothing,AbstractVector} = LinRange(-7.5, 7.5, 200),
+    g::Real = √2,
+    method::WignerSolver = WignerClenshaw(),
+    projection::Union{Val,Symbol} = Val(:two_dim),
+    location::Union{GridPosition,Nothing} = nothing,
+    colorbar::Bool = false,
+    kwargs...,
+) where {DT,OpType<:Union{BraQuantumObject,KetQuantumObject,OperatorQuantumObject}}
+    QuantumToolbox.getVal(projection) == :two_dim ||
+        QuantumToolbox.getVal(projection) == :three_dim ||
+        throw(ArgumentError("Unsupported projection: $projection"))
+
+    return _plot_wigner(
+        library,
+        state,
+        xvec,
+        yvec,
+        QuantumToolbox.makeVal(projection),
+        g,
+        method,
+        location,
+        colorbar;
+        kwargs...,
+    )
+end
+
+function _plot_wigner(
+    ::Val{:CairoMakie},
+    state::QuantumObject{DT,OpType},
+    xvec::AbstractVector,
+    yvec::AbstractVector,
+    projection::Val{:two_dim},
+    g::Real,
+    method::WignerSolver,
+    location::Union{GridPosition,Nothing},
+    colorbar::Bool;
+    kwargs...,
+) where {DT,OpType<:Union{BraQuantumObject,KetQuantumObject,OperatorQuantumObject}}
+    fig, location = _getFigAndLocation(location)
+
+    lyt = GridLayout(location)
+
+    ax = Axis(lyt[1, 1])
+
+    wig = wigner(state, xvec, yvec; g = g, method = method)
+    wlim = maximum(abs, wig)
+
+    kwargs = merge(Dict(:colormap => Reverse(:RdBu), :colorrange => (-wlim, wlim)), kwargs)
+    hm = heatmap!(ax, xvec, yvec, wig'; kwargs...)
```

Comment:
same here, maybe use `transpose` ?

Reply:
I think it is more readable like this. If `wig` is complex, it will return an error. So `adjoint` and `transpose` will always give the same matrix, and when they are not the same, `CairoMakie` will raise an error.

Reply:
So I mean to keep `wig'` instead of `transpose(wig)`, which is more clear.

Reply:
I thought what you need here is actually transpose instead of adjoint, because we need the dimension to match with `xvec` and `yvec`.

Is there an extra reason to add a conjugation here?